src/tasks/common/multimodal_structured.py

```
@@ -0,0 +1,367 @@
+            else:
+                base[key] = value
+
+    def load(self) -> None:
+        """Load dataset, copying from source if needed."""
+        # Check if data exists, otherwise copy from source
+        if not (self.data_dir / "schema.json").exists():
+            if not self.source_path.exists():
+                raise FileNotFoundError(
+                    f"Source directory not found: {self.source_path}\n"
+                    f"Please provide a valid source_dir"
+                )
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)